src/Juvix/Compiler/Core/Extra/Utils.hs

```diff
@@ -352,7 +352,11 @@ translateCase translateIf dflt Case {..} = case _caseBranches of
 
 checkDepth :: Int -> Node -> Bool
 checkDepth 0 _ = False
-checkDepth d node = all (checkDepth (d - 1)) (childrenNodes node)
+checkDepth d node = case node of
+  NApp App {..} ->
+    checkDepth d _appLeft && checkDepth (d - 1) _appRight
```

Comment:
can you explain this to me?

Reply:
Previously e.g. `f a b c` had depth 4 because it's `((f a) b) c`. For the purposes of assessing how big terms are it's more reasonable to treat applications as flat.